src/parsing.js

```diff
@@ -0,0 +1,83 @@
+
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//     You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//     See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function(internal, scope, testing) {
+
+  function isCalc(string) {
+    return /^calc/.test(string); 
+  }
+
+  // Effectively returns a calc dictionary.
+  function parseDimension(unitRegExp, string) {
+    string = string.trim().toLowerCase();
+
+    // CSS allows lengths to be '0', so if px is a supported unit, return 0px.
+    if (string == '0' && 'px'.search(unitRegExp) >= 0)
+      return {px: 0};
+
+    // If we have parenthesis, we're a calc and need to start with 'calc'.
+    if (!/^[^(]*$|^calc/.test(string)) {
+      return;
+    }
+    string = string.replace(/calc\(/g, '(');
```

Comment:
'calc(50px + calc(5em))' is invalid.  However calc(50px + (5em)) is valid.  This line makes both expressions the same.